tests/zfs-tests/tests/functional/tmpfile/tmpfile_002_pos.c

```diff
@@ -0,0 +1,60 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
```

Comment:
Add a block comment describing what operations are being tested by this test case.